### ISSUE TITLE
getUpdateMetadata gives false package with DEBUG build

### ISSUE THREAD

**Issue by adbl** [Microsoft/react-native-code-push#672]
Running a DEBUG build in simulator (and probably on device too), using this in AppDelegate.m:

```
 #ifdef DEBUG
  jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index.ios" fallbackResource:nil];
#else
  jsCodeLocation = [CodePush bundleURL];
#endif
```

which should make it impossible that a Code Push update gets loaded?

1. The first run this works as expected, `codePush.getUpdateMetadata()` returns `null`.
2. During this run, Code Push detected, downloaded and installed an update.
3. During next launch, `codePush.getUpdateMetadata()` now gives the package that was downloaded, however it clearly isn't running. It's running the dev bundle from packager. 

This had me extremely confused until I figured out that it wasn't actually running the update. It seems like `getUpdateMetadata` doesn't understand that an "installed" package may not be running, due to the CodePush bundle loader was not in use.

* react-native-code-push version: `1.16.1-beta`
* react-native version: `0.36.1`
* iOS/Android/Windows version: `iOS 10.2`
* Does this reproduce on a debug build or release build? deug
* Does this reproduce on a simulator, or only on a physical device? Simulator (both?)

**Comment by sergey-akhalkov** [comment]
@adbl - hi!
`getUpdateMetadata` method returns info about latest installed package and supposes that local js code is in use, I don't sure if its behavior should be changed to determine js code location...
To make it easier to understand what js code location is in use we could output its location using this hack:
```
...
const { SourceCode } = require('NativeModules');
...
<Text style={styles.messages}>{SourceCode.scriptURL || ""}</Text>
...
```
Depending on `jsCodeLocation` variable value it prints different paths:
1. `jsCodeLocation = [CodePush bundleURL];` ![image](https://cloud.githubusercontent.com/assets/4928157/22059925/e148298a-dd7f-11e6-9d37-2d21868b3ca4.png)
2. `jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index.ios" fallbackResource:nil];` 
![image](https://cloud.githubusercontent.com/assets/4928157/22059987/1de4f328-dd80-11e6-8024-871fc31efb97.png)

As for debugging: 
``` 
#ifdef DEBUG
  jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index.ios" fallbackResource:nil];
#else
  jsCodeLocation = [CodePush bundleURL]; 

#endif
```
Regarding to this code lines dev bundle is used on each app start. To make it possible to use local js code in debug mode we should done following:
1. Replace `jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index.ios" fallbackResource:nil];` with `jsCodeLocation = [CodePush bundleURL];`.
2. Force the JS bundle to be generated in simulator builds by removing the if block that echoes "Skipping bundling for Simulator platform" in the "node_modules/react-native/packager/react-native-xcode.sh" file.

**Comment by adbl** [comment]
I don't know if it should be changed either, but it did cause confusion for me at least. From that perspective it would be nicer if CodePush could detect that the bundle isn't actually `RUNNING` (which is the default query to `getMetaData`, because it is in fact not running.

If you don't think it should change then I can workaround this by either checking the source url as you suggest and match agains `localhost` or I can simply just check for `__DEV__`. But I prefer to now have to think about `__DEV__` nor js code location. Maybe then, `codePush` could export a new function which let's me know whether React is running code loaded by it?

**Comment by adbl** [comment]
Actually, given the documentation for `getUpdateMetadata`:

> The updateState parameter is set to `UpdateState.RUNNING`, but the app isn't currently running a CodePush update. There may be a pending update, but the app hasn't been restarted yet in order to make it active.

This update could be described as `UpdateState.PENDING`, but that might also be confusing, since codePush cannot know if it _will_ be loaded next time either. Maybe there could be an `UpdateState.INSTALLED` or `UpdateState.INACTIVE` with documentation describing the possible reasons for this update that is installed but still not running?

**Comment by sergey-akhalkov** [comment]
@abdl, I've created https://github.com/Microsoft/react-native-code-push/pull/673 PR contains `isUsingReactNativeBundle` method. Please let me know if  you have any questions or see any issues.

**Comment by adbl** [comment]
Nice @sergey-akhalkov!

Maybe we should also add a bit of documentation to `getUpdateMetadata` to describe the scenario I initially described which had me confused...?

**Comment by sergey-akhalkov** [comment]
@adbl, I've committed docs updates in https://github.com/Microsoft/react-native-code-push/pull/673

**Comment by panarasi** [comment]
I just closed #673 since I am not sure we need to add this method to Code Push. I think a better way would be to update the documentation to specify that only release mode is used. 

@sergey-akhalkov is creating a new bug to update the documentation and will update the bug number here. I think we should close this issue once we have the bug for adding documentation.

**Comment by sergey-akhalkov** [comment]
@panarasi, done! The issue number is #682
